File: utbot-framework/src/main/java/org/utbot/quickcheck/generator/java/time/ZoneIdGenerator.java


package org.utbot.quickcheck.generator.java.time;

import org.utbot.engine.greyboxfuzzer.util.UtModelGenerator;
import org.utbot.framework.plugin.api.UtModel;
import org.utbot.quickcheck.generator.GenerationStatus;
import org.utbot.quickcheck.generator.Generator;
import org.utbot.quickcheck.random.SourceOfRandomness;

import java.time.ZoneId;

import static java.time.ZoneId.getAvailableZoneIds;
import static org.utbot.external.api.UtModelFactoryKt.classIdForType;

/**
 * Produces values of type {@link ZoneId}.
 */
public class ZoneIdGenerator extends Generator<ZoneId> {
    public ZoneIdGenerator() {
        super(ZoneId.class);
    }

    @Override public UtModel generate(
        SourceOfRandomness random,
        GenerationStatus status) {

        return UtModelGenerator.getUtModelConstructor().construct(ZoneId.of(random.choose(getAvailableZoneIds())), classIdForType(ZoneId.class));
    }
}

File: utbot-framework/src/main/kotlin/org/utbot/quickcheck/generator/java/time/ZoneIdGenerator.kt
package org.utbot.quickcheck.generator.java.time

import org.utbot.engine.greyboxfuzzer.util.UtModelGenerator.utModelConstructor
import org.utbot.framework.plugin.api.UtModel
import org.utbot.framework.plugin.api.util.id
import org.utbot.quickcheck.generator.GenerationStatus
import org.utbot.quickcheck.generator.Generator
import org.utbot.quickcheck.random.SourceOfRandomness
import java.time.ZoneId

/**
 * Produces values of type [ZoneId].
 */
class ZoneIdGenerator : Generator(ZoneId::class.java) {
    override fun generate(
        random: SourceOfRandomness,
        status: GenerationStatus
    ): UtModel {
        return utModelConstructor.construct(
            ZoneId.of(random.choose(ZoneId.getAvailableZoneIds())),
            ZoneId::class.id
        )
    }
}